addons/Omen/Omen.lua

```diff
@@ -0,0 +1,287 @@
+            elseif obj_time < 1 and cur < fin then
+                body = body.."\n "..bad_col..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]\\cr"
+            else
+                body = body.."\n "..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]"
+            end
+        end
+    end
+body = string.gsub(body,"%-1","%?%?%?")
+texts.text(IMAGE,header..body)
+end
+
+hide_timer = {"Kin","Gin","Kei","Kyou","Fu","Craver","Gorger","Thinker","Treasure","Waiting"}
+refresh()
+
+
+function test(str)
+windower.add_to_chat(161,tostring(str))
+end
+
+function floor_test()
+windower.add_to_chat(161,"Vanquish 18 sweetwater foes."..string.char(0x7f).."1")
+--windower.add_to_chat(161,"Vanquish the Glassy Gorger."..string.char(0x7f).."1")
+coroutine.sleep(2)
+windower.add_to_chat(161,"You have 90 seconds remaining.")
+coroutine.sleep(2)
+windower.add_to_chat(161,"5: You have reduced your foe's HP by 1500 in a single auto-attack.")
+coroutine.sleep(1)
+windower.add_to_chat(161,"5: You have reduced your foe's HP by at least 2000 in a single auto-attack.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by 1500 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by 5000 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by 14999 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"3: You have reduced your foe's HP by at least 15000 using a single magic attack without performing a magic burst.")
+coroutine.sleep(1)
+windower.add_to_chat(161,"1: Vanquish 6 foes.")
+windower.add_to_chat(161,"2: Use 30 weapon skills on your foes.")
+windower.add_to_chat(161,"3: Reduce your foe's HP by at least 15000 using a single magic attack without performing a magic burst.")
+windower.add_to_chat(161,"4: Use 12 abilities on your foes.")
+windower.add_to_chat(161,"5: Reduce your foe's HP by at least 2000 in a single auto-attack.")
+coroutine.sleep(1)
+windower.add_to_chat(161,"A spectral light flares up."..string.char(0x7f).."1")
+end
+
+
+windower.register_event('prerender', function()
+    if obj_time < 1 then return end
+    if obj_time ~= (end_time - os.time()) then
+        obj_time = end_time - os.time()
+        refresh()
+    end
+end)
+
+windower.register_event('zone change', function(zone)
+    texts.visible(IMAGE,false)
+    floor_obj = "Waiting for objectives..."
+    reset_objectives()
+    if zone == 292 then -- Henge
+        texts.visible(IMAGE,true)
+    end
+end)
+
+texts.visible(IMAGE,false)
+if windower.ffxi.get_info() and windower.ffxi.get_info().zone == 292 then
+texts.visible(IMAGE,true)
+end
+
+windower.register_event('incoming text', function(original, modified, mode)
+    if mode == 161 then -- Omen messages are 161 color, except total time extension messages which are 121 and irrelevant
```

Comment:
When I asked about it on Windower4 thread on BG I got asked that nobody knew which packets exactely Omen used. See reference https://www.bluegartr.com/threads/115131-Windower-4-Discussion-and-Support?p=6860317&viewfull=1#post6860317
Plus like 99,9999% of this addon has been developed by Braden, I only fixed a couple of bugs, added a few things, and my main interested in publishing the addon here is because me and my friends find it really useful, and I think it would be cool to have it on the official windower launcher.
I honestly lack the skill to use packets and even more to reverse engineer to find which bytes exactely are the Omen data.

Aside from these practical limitations and on a theoretical plan I 101% agree with you packets would be the best solution, not a coincidence I asked for those first when the addon didn't exist yet.